app/cmd/root.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright © 2022 Merbridge Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"runtime"
+	"strings"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/merbridge/merbridge/app/cmd/options"
+	"github.com/merbridge/merbridge/config"
+	"github.com/merbridge/merbridge/internal/ebpfs"
+	"github.com/merbridge/merbridge/internal/pods"
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "mbctl",
+	Short: "Use eBPF to speed up your Service Mesh like crossing an Einstein-Rosen Bridge.",
+	Long:  `Use eBPF to speed up your Service Mesh like crossing an Einstein-Rosen Bridge.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		client := options.NewOptions()
+		w := pods.NewWatcher(client)
+
+		_ = w.Start()
+		log.Info("Pod Watcher Ready")
+		ch := make(chan os.Signal, 1)
+		signal.Notify(ch, syscall.SIGTERM, syscall.SIGQUIT, syscall.SIGINT)
+		<-ch
+		w.Stop()
+		_ = ebpfs.UnLoadMBProgs()
+		log.Info("Pod Watcher Down")
+	},
+}
+
+// Execute excute root command and its child commands
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	// Setup log format
+	log.SetFormatter(&log.TextFormatter{
+		DisableTimestamp:       false,
+		FullTimestamp:          true,
+		DisableLevelTruncation: true,
+		DisableColors:          true,
+		CallerPrettyfier: func(f *runtime.Frame) (string, string) {
+			fs := strings.Split(f.File, "/")
+			filename := fs[len(fs)-1]
+			ff := strings.Split(f.Function, "/")
+			_f := ff[len(ff)-1]
+			return fmt.Sprintf("%s()", _f), fmt.Sprintf("%s:%d", filename, f.Line)
+		},
+	})
+	log.SetOutput(os.Stdout)
+	log.SetLevel(log.InfoLevel)
+	log.SetReportCaller(true)
+
+	// Get some flags from commands
+	rootCmd.PersistentFlags().StringVarP(&config.Mode, "mode", "m", config.ModeIstio, "Service mesh mode, current support istio and linkerd")
+	rootCmd.PersistentFlags().BoolVarP(&config.UseReconnect, "use-reconnect", "r", true, "Use re-connect mode as same-node acceleration")
+	rootCmd.PersistentFlags().BoolVarP(&config.Debug, "debug", "d", false, "Debug mode")
+	rootCmd.PersistentFlags().BoolVarP(&config.IsKind, "kind", "k", false, "Kubernetes in Kind mode")
```

Comment:
When this is not very common and the developer needs to be able to clearly understand what he or she is doing, I think we should not use short name.

Reply:
pflags offers the full name and short name at the same time, which means if execute -h, you will see that:
```
Use eBPF to speed up your Service Mesh like crossing an Einstein-Rosen Bridge.

Usage:
  mbctl [flags]

Flags:
  -d, --debug             Debug mode
  -h, --help              help for mbctl
  -f, --ips-file string   Current node ips file name
  -k, --kind              Kubernetes in Kind mode
  -m, --mode string       Service mesh mode, current support istio and linkerd (default "istio")
  -r, --use-reconnect     Use re-connect mode as same-node acceleration (default true)
``

you can use the full name like `--mode istio` and also `-m istio`, if someone is not familiar with the flags they can use full name flag, if someone is familiar with flags, short name is better.